Vagrantfile

```diff
@@ -18,6 +18,34 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   config.vm.synced_folder ".", "/vagrant", disabled: true
   config.vm.synced_folder ".", "/srv/zulip"
 
+  proxy_config_file = ENV['HOME']+"/.zulip-vagrant-config"
+  http_proxy = https_proxy = no_proxy = ""
+
+  if File.file?(proxy_config_file)
+    IO.foreach(proxy_config_file) do |line|
+      line.chomp!
+      key, value = line.split(nil, 2)
+      case key
+      when /^([#;]|$)/; # ignore line
+      when "HTTP_PROXY"; http_proxy = value
+      when "HTTPS_PROXY"; https_proxy = value
+      when "NO_PROXY"; no_proxy = value
+      end
```

Comment:
OK let's stick with the simple custom parser you wrote.  Definitely not
worth adding another dependency, and I think YAML is too error-prone for a
user-configured thing like this.


        -Tim Abbott

On Tue, Apr 12, 2016 at 12:04 PM, Vishnu Ks <notifications@github.com>
wrote:

> In Vagrantfile
> <https://github.com/zulip/zulip/pull/644#discussion_r59434568>:
>
> > @@ -18,6 +18,34 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
> >    config.vm.synced_folder ".", "/vagrant", disabled: true
> >    config.vm.synced_folder ".", "/srv/zulip"
> >
> > +  proxy_config_file = ENV['HOME']+"/.zulip-vagrant-config"
> > +  http_proxy = https_proxy = no_proxy = ""
> > +
> > +  if File.file?(proxy_config_file)
> > +    IO.foreach(proxy_config_file) do |line|
> > +      line.chomp!
> > +      key, value = line.split(nil, 2)
> > +      case key
> > +      when /^([#;]|$)/; # ignore line
> > +      when "HTTP_PROXY"; http_proxy = value
> > +      when "HTTPS_PROXY"; https_proxy = value
> > +      when "NO_PROXY"; no_proxy = value
> > +      end
>
> @timabbott <https://github.com/timabbott> There is a parser
> <https://github.com/chrislee35/configparser> similar to the Python
> Configparser, but the user have to install the gem first before using it.
> Built in support is only available to YAML.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zulip/zulip/pull/644/files/88d1a317589be321cbf2516b4a3548a099369f7a#r59434568>
>